docs/architecture.md

```diff
@@ -0,0 +1,130 @@
+# Fleet Architecture
+
+## Overview
+
+At a Kubernetes level, a fleet consists of a hub cluster and zero or more member clusters.  Custom resources and associated controllers,
+residing both on the hub and member(s), as well as validating webhooks on the hub, govern the behavior of features of the fleet.  This
+document gives an architectural overview of those resources, controllers and validating webhooks.
+
+Fleets are configured such that each member *m* is granted access to resources within its corresponding *fleet-member-m* namespace on the
+hub.  Controllers on both the hub and the member are able to read, write and watch these resources to communicate between each-other.  The
+hub is not granted access to resources on any member, nor does any member have access to any other member.
+
+The following fleet use cases are implemented:
+
+1. **Join/Leave**: member clusters can join and leave a fleet.
+1. **Cluster resource placement (CRP)**: arbitrary Kubernetes resources can be replicated from a hub to one or more members, building upon
```

Comment:
we are deprecating this repo. It's better to refer to the upstream https://github.com/kubernetes-sigs/work-api